qiskit/pulse/instructions/set_frequency.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Set channel frequency.
+"""
+from typing import Optional
+
+from qiskit.pulse.channels import PulseChannel
+from qiskit.pulse.commands.instruction import Instruction
+
+
+class SetFrequency(Instruction):
+    """
+    Set the channel frequency. This command operates on PulseChannels.
+    A PulseChannel creates pulses of the form Re[exp(i 2pi f jdt + phase) d_j].
+    F is the frequency of the channel. The command SetFrequency allows
+    the user to set the value of f. All pulses that are played on a channel
+    after SetFrequency has been called will have the corresponding frequency.
+
+    The duration of SetFrequency is 0.
+    """
+
+    prefix = 'sf'
+
+    def __init__(self, frequency: float,
+                 channel: Optional[PulseChannel] = None,
+                 name: Optional[str] = None):
+        """Creates a new set channel frequency pulse.
+
+        Args:
+            frequency: New frequency of the channel in Hz.
+            channel: The channel this instruction operates on.
+            name: Name of this set channel frequency command.
+        """
+        self._frequency = float(frequency)
+        super().__init__(0, channel, name=name)
+        self._channel = channel
+
+    @property
+    def frequency(self):
+        """New frequency."""
```

Comment:
Maybe this?

```suggestion
        """New channel modulation frequency."""
```

Reply:
For now I want to avoid being too specific here (i.e. avoid the word `modulation`). We will very soon extend `SetFrequency` to be able to change the LO as well as the side-band. Once we extend `SetFrequency` to this (and we have clarified how) I will provide a more detailed docstring.